woocommerce/woocommerce
# Add Tree Select Control Component

### All Submissions:






### Changes proposed in this Pull Request:

This PR adds a new brand component: Tree Select Component.

The component has been already working in WooCommerce ecosystem for 2 months. We used the component in [Google Listing And Ads for WooCommerce](https://github.com/woocommerce/google-listings-and-ads)

So we want now to promote the component to WooCommerce in order to others to be able to use it. 

The component has tests and follows the conventions. It also has their own storybook story.

The original project thread can be found here pcTzPl-fg-p2

### How to test the changes in this Pull Request:

First of all Run `pnpm run storybook` and then enter in http://localhost:6007/?path=/story/woocommerce-admin-components-treeselectcontrol--base in order to check the component in action

1. Check the tree expands on click and you can expand and collapse nodes clicking on the chevron

https://user-images.githubusercontent.com/5908855/176210350-4d27463f-e756-41fe-8c35-783d3521faef.mov

2. Check you can select options and they appear in the tags. Check also you can deselect the option and the tag disappears.


https://user-images.githubusercontent.com/5908855/176210556-c16aa673-5239-4481-97e4-99d1822fea4c.mov

3. Check that clicking on the group selects all the options.

https://user-images.githubusercontent.com/5908855/176210805-2706ad5b-58b4-4b09-b0c5-f4833fdb5f3f.mov

4. Check that when there is more tags than the property `maxVisibleTags` it shows/hide the tags toggler

https://user-images.githubusercontent.com/5908855/176211237-75faf958-9b65-4bc6-ad83-c3d77f37ef93.mov

5. Check that "Select All" label match with the `selectAllLabel` prop. 
6. Check that the placeholder shows when no options are selected and the tree is collapsed. Also it match with the prop `placeholder`
7. Check that the input label match with the `label` prop 
8. Check the a11y. You can move with UP/DOWN and select elements with SPACE, you can expand nodes with RIGHT/LEFT 

9. Check you can filter options searching in the input

https://user-images.githubusercontent.com/5908855/176212080-99a2ab68-cb5d-4e3f-9ccb-c2c4eb7f2b60.mov

10. Check you can import and use the component in WooCommerce correctly. 

### Other information:




### FOR PR REVIEWER ONLY:
